### PR TITLE
PWX-3992: enable tls in consul

### PR DESCRIPTION
**Pull request by sdeoras**
this pr is to enable tls in consul
Signed-off-by: Saurabh Deoras <saurabh@portworx.com>

### CODE REVIEW

File: consul/client.go [hunk 1]
```diff
@@ -182,10 +182,41 @@ func isKeyIndexMismatchErr(err error) bool {
...
-	config.Scheme = "http"
-	config.Token = p.options[kvdb.ACLTokenKey]
+
+	// check if TLS is required
+	caFile, ok1 := p.options[kvdb.CAFileKey]
+	certFile, ok2 := p.options[kvdb.CertFileKey]
+	certKeyFile, ok3 := p.options[kvdb.CertKeyFileKey]
+	caAuthAddress := p.options[kvdb.CAAuthAddress]
+	insecureSkipVerify := strings.ToLower(p.options[kvdb.InsecureSkipVerify]) == "true"
+	if ok1 && ok2 && ok3 {
```
Comment (adityadani) [inline consul/client.go]: In etcd there is a way to setup where only CAFile is required and certFile and certKey is not required. Can consul be setup in such a way? If so we cannot have a hard check on all the ok's here.

File: consul/client.go [hunk 2]
```diff
@@ -182,10 +182,41 @@ func isKeyIndexMismatchErr(err error) bool {
...
+		}
+
+		config := api.DefaultConfig()
+		config.Address = machine
+		config.Scheme = "https"
+		config.HttpClient = new(http.Client)
+
+		config.HttpClient.Transport = &http.Transport{
+			TLSClientConfig: consulTLSConfig,
+		}
```
Comment (adityadani) [inline consul/client.go]: Do we need to set the ACL token key here?

File: consul/client.go (hunk 1 above)
Comment (sdeoras) [reply consul/client.go]: i am not sure. i do have a consul cluster with self signed TLS. i could give it a try there.

File: consul/client.go (hunk 2 above)
Comment (sdeoras) [reply consul/client.go]: i moved the code around to take ACL token input.

**Pull request merged**